### PR TITLE
 Ensure that library addresses supplied are of correct length and hex prefixed

### PR DESCRIPTION
**Pull request by axic**
Depends on #3617 and https://github.com/ethereum/solc-js/pull/201

### CODE REVIEW

File: libsolidity/interface/StandardCompiler.cpp [hunk 1]
```diff
@@ -337,16 +339,30 @@ Json::Value StandardCompiler::compileInternal(Json::Value const& _input)
 			return formatFatalError("JSONError", "library entry is not a JSON object.");
 		for (auto const& library: jsonSourceName.getMemberNames())
 		{
+			string const& address = jsonSourceName[library].asString();
```
Comment (axic) [inline libsolidity/interface/StandardCompiler.cpp]: Need to double check `asString`.

File: test/libsolidity/StandardCompiler.cpp [hunk 2]
```diff
@@ -633,19 +633,15 @@ BOOST_AUTO_TEST_CASE(libraries_invalid_hex)
...
 {
 	char const* input = R"(
 	{
 		"language": "Solidity",
 		"settings": {
 			"libraries": {
 				"library.sol": {
-					"L": 42,
-					"L3": "42",
-					"L4": "0x42",
```
Comment (axic) [inline test/libsolidity/StandardCompiler.cpp:646]: Should leave this test too.

File: libsolidity/interface/StandardCompiler.cpp (hunk 1 above)
Comment (chriseth) [reply libsolidity/interface/StandardCompiler.cpp]: does `asString()` really return a reference?

**Review by chriseth — APPROVED** [review]
Looks good apart from the `const&`

File: libsolidity/interface/StandardCompiler.cpp (hunk 1 above)
Comment (axic) [reply libsolidity/interface/StandardCompiler.cpp]: No, it actually creates a new instance in some cases.

**Pull request merged**